third_party/amd/backend/compiler.py

```diff
@@ -23,7 +23,7 @@ class HIPOptions:
     arch: str = None
     allow_fp8e4nv: bool = False
     default_dot_input_precision: str = "ieee"
-    allowed_dot_input_precisions: Tuple[str] = ("ieee", )
+    allowed_dot_input_precisions: Tuple[str] = ("tf32", "ieee")
```

Comment:
@zhanglx13 
I have a question about this part.
I see two possibilities here:

1. assume that TF32 is optional low-precision mode of float32, so we can use ordinary float32 even if TF32 is set (this is what happens in this PR)
2. TF32 is not supported by AMD backend and related tests should be simply skipped

Reply:
From what I saw in `test_dot`, tf32 is skipped So I assume it's ok to just skip them on AMD backend.